Core/Security/VOMSService.py

```diff
@@ -1,46 +1,78 @@
-# $HeadURL$
+""" VOMSService class encapsulated connection to the VOMS service for a given VO
+"""
+
 __RCSID__ = "$Id$"
 
 from DIRAC import gConfig, S_OK, S_ERROR
 from DIRAC.Core.Utilities.SOAPFactory import getSOAPClient
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getVOOption
+from DIRAC.ConfigurationSystem.Client.Helpers.CSGlobals import getVO
 
 def _processListReturn( soapReturn ):
   data = []
-  for entry in soapReturn:
-    data.append( str( entry ) )
+  if soapReturn:
+    for entry in soapReturn:
+      data.append( str( entry ) )
   return data
 
 def _processListDictReturn( soapReturn ):
   data = []
-  for entry in soapReturn:
-    entryData = {}
-    for info in entry:
-      entryData[ info[0] ] = str( info[1] )
-    data.append( entryData )
+  if soapReturn:
+    for entry in soapReturn:
+      entryData = {}
+      for info in entry:
+
+        try:
+          entryData[ info[0] ] = str( info[1] )
+        except:
+          pass
+      data.append( entryData )
   return data
 
-class VOMSService:
+class VOMSService( object ):
+
+  def __init__( self, vo = None, adminUrl = False, attributesUrl = False, certificatesUrl = False ):
+
+    if vo is None:
+      vo = getVO()
+    if not vo:
+      raise Exception( 'No VO name given' )
 
-  def __init__( self, adminUrl = False, attributesUrl = False, certificatesUrl = False ):
+    self.vo = vo
+    self.vomsVO = getVOOption( vo, "VOMSName" )
+    if not self.vomsVO:
+      raise Exception( "Can not get VOMS name for VO %s" % vo )
     self.__soapClients = {}
     for key, url in ( ( 'Admin', adminUrl ), ( 'Attributes', attributesUrl ), ( 'Certificates', certificatesUrl ) ):
+      urls = []
       if not url:
-        url = gConfig.getValue( "/Registry/VOMS/URLs/VOMS%s" % key, "" )
+        url = gConfig.getValue( "/Registry/VO/%s/VOMSServices/VOMS%s" % ( self.vo, key ), "" )
```

Comment:
Why ? It is correct, look at the LHCb CS, for example. The VOMS section is obsoleted. To keep the same arrangement for single and multi VO installations